pkgs/applications/misc/gpsbabel/default.nix

```
@@ -1,59 +1,97 @@
-{ lib, stdenv, fetchFromGitHub, fetchpatch, libusb1, qtbase, zlib, IOKit, which, expat }:
+{ lib, stdenv, fetchFromGitHub, fetchurl, pkg-config, which
+, qtbase, qmake, qttools, qttranslations, wrapQtAppsHook
+, libusb1, shapelib, zlib
+, withGUI ? false, qtserialport
+, withMapPreview ? (!stdenv.isDarwin), qtwebengine
+, withDoc ? false, docbook_xml_dtd_45, docbook_xsl, expat, fop, libxml2, libxslt, perl
```

```suggestion
, docbook_xml_dtd_45, docbook_xsl, expat, fop, libxml2, libxslt, perl
```

we don't need this flag if we use a separate output.

---

Without the ability to turn the build of the docs off, architectures with a broken docs dependency would also lose this package.

---

Though agreed the docs should have a separate output

---

gpsbabel (non-GUI) often used as runtime dependency (viking, pytrainer, etc), in such case usually we don't need docs and don't want to build it. As far as I know there is no way to skip docs building with separate output, but without this option, am I right?

---

Could you not do `outputs = [ "out" ] ++ lib.optionalString withDoc "doc";` ?

---

Sure, fixed.

---

> Without the ability to turn the build of the docs off, architectures with a broken docs dependency would also lose this package.

With the exception of fop all of these dependencies are super important and a build failure for one of them would cause a big fallout in nixpkgs.

---

IDK why we need the option to not build the docs. Does it take multiple minutes to build? If not then IMO it is just added complexity.

---

It takes ~1min to build documentation on my machine.

---

What about building documentation only for `gpsbabel-gui`? To keep `gpsbabel` minimal.